src/plugins/data/public/ui/dataset_selector/configurator.tsx

```diff
@@ -121,6 +121,7 @@ export const Configurator = ({
                   setTimeFieldName(value);
                   setDataset({ ...dataset, timeFieldName: value });
                 }}
+                disabled={dataset?.timeFieldName !== undefined}
```

Comment:
does this disable the moment the user switches the timeFieldName?

for dataset.type === `INDEX_PATTERN` did we want to consider just hiding this select? but i haven't verified that we are updating the cache correctly if the user wants to update the timefield for non index patterns. so perhaps we should consider letting them set this even if the data set is created and hiding it in the case of index_pattern.

Reply:
for some reason cant seem to set `No time field`  to be undefined for latest main. It is setting it to `No time field` can we verify this?

No time field should map to undefined and time field should should then be undefined.

Reply:
> Or disable only type index pattern
@ashwin-pc also suggested this, updated in 3b77d5b. I think it'll be helpful to users to show whether or not there is already a time field assigned to the index pattern. Also, if there are no date fields, then the selector won't show at all.

> for some reason cant seem to set No time field to be undefined for latest main. It is setting it to No time field can we verify this?
No time field should map to undefined and time field should should then be undefined.

It works for me on `main`; I'm able to set `No time field` to undefined and it'll load correctly. One thing to note is that for PPL, we may need to update the search interceptor/strategy since it'll automatically append a `timefield` clause, which will break if we select no time field.